s3/replication/common/src/s3replicationcommon/jobs.py

```diff
@@ -264,8 +264,18 @@ def add_job(self, job):
         self._jobs[job.get_replication_id()] = job
         self._job_id_to_replication_id_map[job.get_job_id()] = \
             job.get_replication_id()
-        # Initial state is queued.
-        self._jobs_queued[job.get_replication_id()] = None
+
+        if self._label == "completed-jobs":
+            # As the state of job is changed to completed, add the job in
+            # jobs_completed
+            self._logger.debug("Jobs[{}]: Adding job with job_id {} into jobs_completed".
+                               format(self._label, job.get_job_id()))
+            self._jobs_completed.add(job.get_replication_id())
+        else:
+            # Initial state is queued.
```

Comment:
Initial state is queued for "all-jobs" type object.

Reply:
done